app/models/audit/AuditTaskInteractionTable.scala

```diff
@@ -261,6 +261,218 @@ object AuditTaskInteractionTable {
+    ).first
+  }
+
+  /**
+    * Calculates the median auditing time per 100 meters for all users who have audited
+    * at least 100 meters and spent 30 minutes auditing.
+    */
+  def countTotalAverageAuditingTime(): Float = db.withSession { implicit session =>
+    Q.queryNA[Float](
+      s"""SELECT percentile_CONT(0.5) WITHIN GROUP (ORDER BY minutes_per_100m)
+         |FROM (
+         |    SELECT user_stat.user_id,
+         |           minutes_audited / (meters_audited / 100) AS minutes_per_100m
+         |    FROM user_stat
+         |    INNER JOIN (
+         |        SELECT user_id,
+         |               CAST(
+         |                 extract(second from SUM(diff)) / 60 +
+         |                 extract(minute from SUM(diff)) +
+         |                 extract(hour from SUM(diff)) * 60 AS decimal(10,2)) AS minutes_audited
+         |        FROM (
+         |            SELECT audit_task.user_id, timestamp,
+         |                   (timestamp - LAG(timestamp, 1) OVER(PARTITION BY audit_task.user_id ORDER BY timestamp)) AS diff
+         |            FROM audit_task_interaction_small
+         |            INNER JOIN audit_task ON audit_task_interaction_small.audit_task_id = audit_task.audit_task_id
+         |        ) "time_diffs"
+         |        WHERE diff < '00:00:30.000' AND diff > '00:00:00.000'
+         |        GROUP BY user_id
+         |    ) "audit_times" ON user_stat.user_id = audit_times.user_id
+         |    WHERE meters_audited > 100
+         |        AND minutes_audited > 30
+         |) AS filtered_data;
+         |""".stripMargin
+    ).first
+  }
+
+  /**
+    * Calculates the median auditing time per 100 meters for all users in the past 24 hours
+    * who have audited at least 100 meters and spent 30 minutes auditing.
+    */
+  def countTodayAverageAuditingTime(): Float = db.withSession { implicit session =>
+    Q.queryNA[Float](
+      s"""SELECT percentile_CONT(0.5) WITHIN GROUP (ORDER BY minutes_per_100m)
+         |FROM (
+         |    SELECT user_stat.user_id,
+         |           minutes_audited / (meters_audited / 100) AS minutes_per_100m
+         |    FROM user_stat
+         |    INNER JOIN (
+         |        SELECT user_id,
+         |               CAST(
+         |                 extract(second from SUM(diff)) / 60 +
+         |                 extract(minute from SUM(diff)) +
+         |                 extract(hour from SUM(diff)) * 60 AS decimal(10,2)) AS minutes_audited
+         |        FROM (
+         |            SELECT audit_task.user_id, timestamp,
+         |                   (timestamp - LAG(timestamp, 1) OVER(PARTITION BY audit_task.user_id ORDER BY timestamp)) AS diff
+         |            FROM audit_task_interaction_small
+         |            INNER JOIN audit_task ON audit_task_interaction_small.audit_task_id = audit_task.audit_task_id
+         |            WHERE (timestamp AT TIME ZONE 'US/Pacific') > (NOW() AT TIME ZONE 'US/Pacific')
```

Comment:
Same issue here, data will always be null because we are only looking at data from the future! lol